bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HostnameKeyStoreMapping.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import java.security.KeyStore;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Provides a TrustManager for supported endpoints
+ *
+ * @author Martin van Wingerden - Initial Contribution
+ */
+@NonNullByDefault
+public interface HostnameKeyStoreMapping {
```

Comment:
For being the major public extension point (interface) that add-on developers are expected to implement, this is imho a very cryptic name and also does not indicate at all that this is expected to be a service to be registered. How about something like `TLSCertificateProvider` or similar?